File: ObjectAvoidance/java/src/main/java/elite/math/trig/triangle/Hypotenuse.java
package elite.math.trig.triangle;

import elite.math.SquareRoot;

/**
 *
 * @author CFerg (Elite)
 */
public class Hypotenuse {

    /**
     *
     */
    public double hypotenuse;
    
    /**
     *
     */
    public Hypotenuse(){
        this.hypotenuse = 0.0;
    }
    
    /**
     *
     * @param a
     * @param o
     */
    public Hypotenuse(Adjacent a, Opposite o){
        this.hypotenuse = new SquareRoot((a.adjacent * a.adjacent) + (o.opposite * o.opposite)).squareRoot;
    }
}

File: ObjectAvoidance/src/elite/math/trig/triangle/Hypotenuse.kt
package elite.math.trig.triangle

import elite.math.SquareRoot

/**
 *
 * @author CFerg (Elite)
 */
class Hypotenuse {

    /**
     *
     */
    var hypotenuse: Double = 0.toDouble()

    /**
     *
     */
    constructor() {
        this.hypotenuse = 0.0
    }

    /**
     *
     * @param a
     * @param o
     */
    constructor(a: Adjacent, o: Opposite) {
        this.hypotenuse = SquareRoot(a.adjacent * a.adjacent + o.opposite * o.opposite).squareRoot
    }
}